l10n_ch_base_bank/invoice.py

```diff
@@ -103,7 +103,8 @@ def _check_bvr(self, cr, uid, ids, context=None):
         """
         invoices = self.browse(cr, uid, ids)
         for invoice in invoices:
-            if invoice.reference_type == 'bvr' and invoice.state != 'draft':
+            check_state = invoice.state not in ('draft', 'cancel')
```

Comment:
Wouldn't it be more readable to skip check, by writing:

    if invoice.state in ('draft', 'cancel'):
        continue

It's certainly only a matter of taste.